### PR TITLE
Extend pages with inherited members

### LINKED ISSUE #190: Extend pages with inherited fields and types
Currently we add only inherited methods

### PR DESCRIPTION
**Pull request by pikinier20**
closes #190

### CODE REVIEW

File: src/main/scala/dotty/dokka/tasty/SyntheticSupport.scala [hunk 1]
```diff
@@ -70,6 +72,24 @@ trait SyntheticsSupport:
     val sym = rsym.asInstanceOf[dotc.core.Symbols.Symbol]
     sym.is(dotc.core.Flags.Extension)
   }
+  /* We need there to filter out symbols with certain flagsets, because these symbols come from compiler and TASTY can't handle them well. 
+  They are valdefs that describe case companion objects and cases from enum.
+  TASTY crashed when calling _.tree on them.
```
Comment (romanowski) [inline src/main/scala/dotty/dokka/tasty/SyntheticSupport.scala:77]: @abgruszecki should we reported it as a bug in TASTY?

**Review by romanowski — APPROVED** [review]
I think we can merge it as long as @abgruszecki approves the hack we've needed to introduce.

File: src/main/scala/dotty/dokka/tasty/SyntheticSupport.scala (hunk 1 above)
Comment (abgruszecki) [reply src/main/scala/dotty/dokka/tasty/SyntheticSupport.scala:77]: We get these symbols by hacking into the compiler, so definitely not. OTOH we may extend Tasty with some functionality based on what we needed to hack into in the doctool, but that's basically a separate issue.

File: src/main/scala/dotty/dokka/tasty/ClassLikeSupport.scala [hunk 2]
```diff
@@ -82,32 +130,53 @@ trait ClassLikeSupport:
...
+
+    def getMethods: List[Symbol] = extractMethods(c.symbol.classMethods)
+
+    def getInheritedDefinitions: InheritedDefinitions = {
+      val trees = c.inheritedMembers
+      .filter(s => s.maybeOwner != defn.ObjectClass && s.maybeOwner != defn.AnyClass)
+      .map(_.tree)
+      
+      val inheritedDefDefs = trees
+        .collect { case dd: DefDef if dd.name != "<init>" => dd.symbol }
```
Comment (abgruszecki) [inline src/main/scala/dotty/dokka/tasty/ClassLikeSupport.scala]: It's better to instead check if `dd` is a constructor, one can in principle define a method `` def `<init>` ``.

File: src/main/scala/dotty/dokka/tasty/SyntheticSupport.scala [hunk 3]
```diff
@@ -70,6 +72,24 @@ trait SyntheticsSupport:
...
+    given ctx as dotc.core.Contexts.Context = r.rootContext.asInstanceOf
+    val sym = rsym.asInstanceOf[dotc.core.Symbols.Symbol]
+    sym.typeRef.appliedTo(sym.typeParams.map(_.typeRef)).allMembers.iterator.map(_.symbol)
+      .collect {
+
+         case sym if  
+          !sym.is(dotc.core.Flags.ModuleVal) && 
+          !(sym.is(dotc.core.Flags.Enum) && sym.is(dotc.core.Flags.Case) && sym.is(dotc.core.Flags.JavaStatic)) => sym 
+        }
+      .map(_.asInstanceOf[r.Symbol])
```
Comment (abgruszecki) [inline src/main/scala/dotty/dokka/tasty/SyntheticSupport.scala]: Better written as:
```suggestion
       .collect {
         case sym if
          !sym.is(dotc.core.Flags.ModuleVal) &&
          !sym.flags.isAllOf(dotc.core.Flags.Enum | dotc.core.Flags.Case | dotc.core.Flags.JavaStatic) =>
              sym.asInstanceOf[r.Symbol]
        }
```

File: src/main/scala/example/Inheritance.scala [hunk 4]
```diff
@@ -7,3 +7,10 @@ abstract class DocumentationInheritance[T, A <: Int, B >: String, -X, +Y] extend
 class DocumentationInheritanceMethod:
     def wierdMethod[T, A <: Int, B >: String](t: T, a: A): B = ???
     def threOtherWay[A <: Nothing, B >: Any](a: A, c: B): Unit = ???
+
+class A:
+    class B extends C:
+        class D extends C
```
Comment (abgruszecki) [inline src/main/scala/example/Inheritance.scala]: More tests would be good:
```suggestion
class A:
    type I = Int
    given Unit = ()
    extension (u: Unit) def foo = "foo"
    object X
    class B extends C:
        class D extends C
```

**Review by abgruszecki — COMMENTED** [review]
LGTM, modulo the comments.

File: src/main/scala/dotty/dokka/tasty/ClassLikeSupport.scala (hunk 2 above)
Comment (pikinier20) [reply src/main/scala/dotty/dokka/tasty/ClassLikeSupport.scala]: I saw below that we check if dd is constructor by checking its name and followed example

File: src/main/scala/dotty/dokka/tasty/SyntheticSupport.scala (hunk 3 above)
Comment (pikinier20) [reply src/main/scala/dotty/dokka/tasty/SyntheticSupport.scala]: changed

File: src/main/scala/example/Inheritance.scala (hunk 4 above)
Comment (pikinier20) [reply src/main/scala/example/Inheritance.scala]: changed

**Review by abgruszecki — APPROVED** [review]
LGTM!

File: src/main/scala/dotty/dokka/tasty/ClassLikeSupport.scala (hunk 2 above)
Comment (abgruszecki) [reply src/main/scala/dotty/dokka/tasty/ClassLikeSupport.scala]: I changed both to `.isClassConstructor`.

**Pull request merged**